Zend/zend_constants.c

```diff
@@ -330,6 +343,14 @@ ZEND_API zval *zend_get_constant_ex(zend_string *cname, zend_class_entry *scope,
 			}
 		}
 
+		if (!current_scope) {
+			if (EG(current_execute_data)) {
```

Comment:
Not sure, but probably `current_execute_data` will be set during compilation as well. Better check `in_compilation`?